### PR TITLE
add support for running pflake8 instead of flake8

### PR DESCRIPTION
**Pull request by jdkandersson**
The linting checks assume `flake8` is used when many of our projects use `pflake8`, this PR adds support for either `flake8` of `pflake8`. It also is more flexible with reading tox output to support more versions of `tox` (the previous regex wasn't going to work with the latest `tox` version released)

### CODE REVIEW

**Review by yanksyoon — DISMISSED** [review]
LGTM!

**Review by arturo-seijas — DISMISSED** [review]
LGTM

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -210,15 +210,15 @@ jobs:
               "
           for EXPECTED_LINT_DEP in $EXPECTED_LINT_DEPS; do
               # Check that the `lint: ` includes the expected dependency
```
Comment (gregory-schiano) [inline .github/workflows/test.yaml:212]: Should we review the comment based on the chage ?

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -230,7 +230,7 @@ jobs:
               "
           for EXPECTED_LINT_CMD in $EXPECTED_LINT_CMDS; do
               # Check that there was a `lint run-test: ` line for the expected command
```
Comment (gregory-schiano) [inline .github/workflows/test.yaml:232]: Same comment concerning the comment :)

File: .github/workflows/test.yaml (hunk 1 above)
Comment (jdkandersson) [reply .github/workflows/test.yaml]: oops, fixed, thanks

File: .github/workflows/test.yaml (hunk 2 above)
Comment (jdkandersson) [reply .github/workflows/test.yaml]: oops, fixed, thanks

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Comment by jdkandersson** [comment]
Had to merge main branch changes, need another round of reviews

**Pull request merged**